### DISCUSSION TITLE
Search Query - Metadata?

### DISCUSSION THREAD

**Discussion by Gareth064** [pnp/pnpcore#1084] [Q&A]
I want to be able to search via a documents metadata, but getting no results.
I am using this query `"ecmUniqueDocumentId: \"231fe9ba-782a-4e86-820f-3f4dcf425461\""`

ecmUniqueDocumentId is a metadata field on a content type used on our documents.

I have also tried '"ecmUniqueDocumentId: 231fe9ba-782a-4e86-820f-3f4dcf425461"' which is wrapping the value in double quotes and that isn't working either.

BUT if I just query `"231fe9ba-782a-4e86-820f-3f4dcf425461"` on its own, I get results.

Now, this value is  GUID so it should only ever be on a single document in our whole tenant... but why can't I search using the metadata field name where this value is stored against?

**Comment by jansenbe** [comment]
Did you create a managed property for the crawled `ecmUniqueDocumentId` property or map to another existing managed property?

**Reply by Gareth064** [reply]
Now you're talking in riddles to me 😄 Unfortunately I don't know much about SPO myself. I am just writing an integration service which allows other services to upload and retrieve documents from SPO.

I will ask your question to the SPO team to see what they say.

**Reply by Gareth064** [reply]
Morning. Ok so I got this back from our SPO team.

![image](https://user-images.githubusercontent.com/22307223/216270466-f071314c-b0d0-4f1f-be6e-0bfcfc76e5dd.png)

![image](https://user-images.githubusercontent.com/22307223/216270725-790a27bb-507c-4b31-8749-906b72553081.png)

I tried using `ECMUNIQUEID` & `OWS_ECMUNIQUEDOCUMENTID` as the query but still nothing.

Is it not working because the Search value isn't set to anything?

Also another thing I just thought about. Search required me to get permission `SharePoint.Sites.Read.All` added to my app reg, and in my head I think that means that it will Search all sites using the query I passed.... but because I need to start a context on a particular site, does that mean it will **only** search that site?

**Reply by Gareth064** [reply]
Looks like we can't tick the Searchable box on a RefinableString?

**Reply by jansenbe** [reply]
Think the query part is enough for searching. After updating managed properties it will take some time to reindex the content again, you can force that for the library in question by going to library settings --> advanced settings --> 

![image](https://user-images.githubusercontent.com/7451219/216306870-d717373e-bd7b-465a-82b3-64a344c211cc.png)

Regarding scope the search to a single site, this is not automatic based upon the context, you need to adjust the query for that by adding 

`path:"https://contoso.sharepoint.com/sites/asite"`

**Reply by jansenbe** [reply]
FYI: https://sharepointmaven.com/how-to-create-a-managed-property-in-sharepoint-online/ is a useful resource

**Reply by Gareth064** [reply]
the SPO team said these managed properties have been setup for months so they would have been indexed every night.

So still not able to get it to work searching against that Column/Metadata.

**Reply by jansenbe** [reply]
Please give the re-index doc library a try and then, after some time,  test for that library. Also, did you try this `ecmUniqueDocumentId: 231fe9ba-782a-4e86-820f-3f4dcf425461`  , so no quotes and no spaces
